### PR TITLE
Return same instance of state if no changes are made in mapReducer

### PR DESCRIPTION
**Pull request by vedrani**

### CODE REVIEW

File: test/mapReducers.spec.js [hunk 1]
```diff
@@ -80,6 +80,23 @@ describe('Map reducers', () => {
     expect(state).to.deep.equal({ 5: { data: 'today' } });
   });
 
+  it('returns state same instance if no change is applied', () => {
```
Comment (zrumenjak) [inline test/mapReducers.spec.js]: same state instance

File: src/mapReducers.js [hunk 2]
```diff
@@ -26,6 +26,40 @@ function resolveKey(action, keySelector) {
   return resolvedKey || TARGET_ALL_REDUCERS;
 }
 
+function calculateNewTargetAllState(state, action, resolveReducer) {
```
Comment (zrumenjak) [inline src/mapReducers.js]: Maybe `calculateNewStateForAllKeys`, and I would use the `calculateNewStateForKey` to perform this operation for each key. You could then do the `shallowEquals` on the resulting state to determine if it has changed (`_.eq` should be good enough for that).

Comment (zrumenjak) [reply src/mapReducers.js]: Also, document that we are doing this in order to preserve the old references when the state doesn't change.

**Review by zrumenjak — APPROVED** [review]

File: src/mapReducers.js (hunk 2 above)
Comment (vedrani) [reply src/mapReducers.js]: You approved, and I hope I fixed it right. Pls comment if not.

**Pull request merged**